Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp

```diff
@@ -358,6 +358,11 @@ static gboolean wpeViewWaylandRenderBuffer(WPEView* view, WPEBuffer* buffer, con
     if (!wlBuffer)
         return FALSE;
 
+    if (wpe_view_get_width(view) != wpe_buffer_get_width(buffer) || wpe_view_get_height(view) != wpe_buffer_get_height(buffer)) {
+        wpe_view_buffer_rendered(view, buffer);
```

Comment:
This might be problematic too, we are expected to reply right after the next vblank.

Reply:
While it's true that you're correct in theory, this particular case is very specific. Window resizing is an occasional event that happens quite rarely and indeed the user it can expect some discontinuity in the frame rate sync.

To meet your expectations, I’m thinking we could keep calling to `wpe_view_buffer_rendered()` inside a `wl_surface_frame()` listener but avoid attaching the buffer to the `surface` using `wl_surface_attach()`. That means something like _do nothing but wait for the next tick_. 

That would preserve the frame throttling hint. However, it seems to me like overcomplicating things for a sporadic situation.

Reply:
I see, while the window is being resized we can receive a buffer for the previous size. In that case it's probably ok to just skip immediately since we have already requested another one for the current size. The question now is, why is this specific to wayland? I wonder if should handle this at AcceleratedBackingStoreDMABuf and simply not call wpe_view_render_buffer()

Reply:
And call frameDone() of course, to notify the web process

Reply:
My understanding is Wayland (more specific) the Wayland compositor (in this case Weston) performs a check in the server side. Unfortunately this check is not just a warning but aborts all the rendering. I don't know how others backends deal with this situation. Are you thinking in GTK? I don't know to be honest.